core/src/components/popover/test/basic/popover.e2e.ts

```diff
@@ -38,145 +38,124 @@ test.describe('popover: focus trap', async () => {
   });
 
   test('should focus the first ion-item on ArrowDown', async ({ page }) => {
+    const item0 = page.locator('ion-popover ion-item:nth-of-type(1)');
+
     await openPopover(page, 'basic-popover');
 
     await page.keyboard.press('ArrowDown');
-    await expectActiveElementTextToEqual(page, 'Item 0');
+    await expect(item0).toBeFocused();
   });
 
   test('should trap focus', async ({ page, browserName }) => {
-    await openPopover(page, 'basic-popover');
-
-    await page.keyboard.press('Tab');
+    const tabKey = browserName === 'webkit' ? 'Alt+Tab' : 'Tab';
+    const items = page.locator('ion-popover ion-item');
 
-    await expectActiveElementTextToEqual(page, 'Item 0');
-
-    await page.keyboard.down('Shift');
-    if (browserName === 'webkit') {
-      await page.keyboard.down('Alt');
-    }
-    await page.keyboard.press('Tab');
-    await page.keyboard.up('Shift');
-    if (browserName === 'webkit') {
-      await page.keyboard.up('Alt');
-    }
+    await openPopover(page, 'basic-popover');
 
-    await expectActiveElementTextToEqual(page, 'Item 3');
+    await page.keyboard.press(tabKey);
+    await expect(items.nth(0)).toBeFocused();
 
-    await page.keyboard.press('Tab');
+    await page.keyboard.press(`Shift+${tabKey}`);
+    await expect(items.nth(3)).toBeFocused();
 
-    await expectActiveElementTextToEqual(page, 'Item 0');
+    await page.keyboard.press(tabKey);
+    await expect(items.nth(0)).toBeFocused();
 
     await page.keyboard.press('ArrowDown');
-
-    await expectActiveElementTextToEqual(page, 'Item 1');
+    await expect(items.nth(1)).toBeFocused();
 
     await page.keyboard.press('ArrowDown');
-
-    await expectActiveElementTextToEqual(page, 'Item 2');
+    await expect(items.nth(2)).toBeFocused();
 
     await page.keyboard.press('Home');
-
-    await expectActiveElementTextToEqual(page, 'Item 0');
+    await expect(items.nth(0)).toBeFocused();
 
     await page.keyboard.press('End');
-
-    await expectActiveElementTextToEqual(page, 'Item 3');
+    await expect(items.nth(3)).toBeFocused();
   });
 
   test('should not override keyboard interactions for textarea elements', async ({ page, browserName }) => {
+    const tabKey = browserName === 'webkit' ? 'Alt+Tab' : 'Tab';
+    const innerNativeTextarea = page.locator('ion-textarea textarea');
+    const vanillaTextarea = page.locator('ion-textarea + textarea');
+
     await openPopover(page, 'popover-with-textarea');
+
+    /**
+     * Focusing happens async inside of popover so we need
+     * to wait for the requestAnimationFrame to fire.
+     */
     await page.waitForFunction(() => document.activeElement?.tagName === 'ION-POPOVER');
```

Comment:
Are we able to use `toBeFocused` on the popover instead?

Reply:
Looks like that does the trick!